catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/web/metrics/IcebergMetricsManager.java

```diff
@@ -0,0 +1,169 @@
+/*
+ *  Copyright 2023 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.catalog.lakehouse.iceberg.web.metrics;
+
+import com.datastrato.gravitino.catalog.lakehouse.iceberg.IcebergConfig;
+import com.datastrato.gravitino.catalog.lakehouse.iceberg.ops.IcebergTableOps;
+import com.datastrato.gravitino.catalog.lakehouse.iceberg.web.IcebergRestUtils;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.iceberg.metrics.MetricsReport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IcebergMetricsManager {
+  private static final Logger LOG = LoggerFactory.getLogger(IcebergTableOps.class);
+  public static final String ICEBERG_METRICS_STORAGE = "metrics-storage";
+  public static final String ICEBERG_METRICS_STORAGE_RETAIN_DAYS = "metrics-storage-retain-days";
+
+  enum MetricsStorageType {
+    EMPTY
+  };
+
+  private final IcebergMetricsFormatter icebergMetricsFormatter;
+  private final IcebergMetricsStorage storage;
+  private final boolean asyncMode = true;
+  private final BlockingQueue<MetricsReport> queue;
+  private final Thread metricsWriterThread;
+  private final int retainDays;
+
+  private volatile boolean isClosed = false;
+  private ScheduledExecutorService scheduledThreadPoolExecutor;
+
+  public IcebergMetricsManager(IcebergConfig icebergConfig) {
+    icebergMetricsFormatter = new IcebergMetricsFormatter();
+    String metricsStorage = icebergConfig.get(IcebergConfig.ICEBERG_METRICS_STORAGE);
+    if (StringUtils.isBlank(metricsStorage)) {
+      metricsStorage = "EMPTY";
+    }
+    MetricsStorageType type = MetricsStorageType.valueOf(metricsStorage.toUpperCase());
+    LOG.info("Iceberg metrics storage: {}", type);
+    switch (type) {
+      case EMPTY:
+        storage = new EmptyMetricsStorage();
+        break;
+      default:
+        throw new RuntimeException("Unknown Iceberg metrics storage: " + type);
+    }
+
+    retainDays = icebergConfig.get(IcebergConfig.ICEBERG_METRICS_STORAGE_RETAIN_DAYS);
+    if (retainDays > 0) {
+      scheduledThreadPoolExecutor =
+          new ScheduledThreadPoolExecutor(
+              1,
+              new ThreadFactoryBuilder()
+                  .setDaemon(true)
+                  .setNameFormat("Iceberg-metrics-cleaner")
+                  .setUncaughtExceptionHandler(
+                      (t, e) -> LOG.error("Uncaught exception in thread {}", t, e))
+                  .build());
+    }
+
+    if (asyncMode) {
```

Comment:
What if the asyncMode is `false`?

Reply:
if `asyncMode` is `false`, metrics will be stored by the specified storage directly.